src/core/mount.c

```diff
@@ -646,9 +646,6 @@ static int mount_add_extras(Mount *m) {
 
         if (!m->where) {
                 r = unit_name_to_path(u->id, &m->where);
-                if (r == -ENAMETOOLONG)
-                        log_unit_error_errno(u, r, "Failed to derive mount point path from unit name, because unit name is hashed. "
-                                                   "Set \"Where=\" in the unit file explicitly.");
```

Comment:
hmm? unit_name_to_path is defined like this:

```c
int unit_name_to_path(const char *name, char **ret) {
        _cleanup_free_ char *prefix = NULL;
        int r;

        assert(name);

        r = unit_name_to_prefix(name, &prefix);
        if (r < 0)
                return r;

        if (unit_name_is_hashed(name))
                return -ENAMETOOLONG;

        return unit_name_path_unescape(prefix, ret);
}
```

so it definitely can return ENAMETOOLONG?

Reply:
I guess the confusion here is between unit_name_to_path() and unit_name_from_path(). Only the latter stopped generating ENAMETOOLONG...